pages/learn/miscellaneous-examples.md

```diff
@@ -157,3 +174,243 @@ We also introduce the following with this example:
+    },
+    "hobbies": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": ["name", "age"]
+}
+```
+
+**Data**
+
+```json
+{
+  "name": "John Doe",
+  "age": 25,
+  "address": {
+    "street": "123 Main St",
+    "city": "New York",
+    "state": "NY",
+    "postalCode": "10001"
+  },
+  "hobbies": ["reading", "running"]
+}
+```
+
+The provided data conforms to the schema by including values for the required properties and ensuring the `age` is an integer greater than or equal to zero. The `address` object contains all the necessary properties, and the `hobbies` property is an array of strings.
+
+
+## Conditional validation with dependencies
+
+The schema below defines an object with two properties: `isMember` and `membershipNumber`. The `membershipNumber` property is required and must have a length of 10 characters only when the `isMember` property is set to true.
+
+```json
+{
+  "$id": "https://example.com/conditional-validation.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Conditional Validation",
+  "type": "object",
+  "properties": {
+    "isMember": {
+      "type": "boolean"
+    },
+    "membershipNumber": {
+      "type": "string",
+      "maxLength": 10,
+      "minLength": 10,
+      "dependencies": {
+        "isMember": true
+      }
+    }
+  }
+}
+```
+
+**Data**
+
+```json
+{
+  "isMember": true,
+  "membershipNumber": "1234567890"
+}
+```
+
+The `dependencies` keyword is used to enforce the presence of the `membershipNumber` property when `isMember` is set to true. The provided data satisfies these conditions.
+
+
+## Conditional validation with if-else
+
+In this schema, we have two properties: `isMember` and `membershipNumber`. The conditional validation is based on the value of the `isMember` property. The validation keywords [if](https://json-schema.org/draft/2020-12/json-schema-core.html#section-10.2.2.1), [then](https://json-schema.org/draft/2020-12/json-schema-core.html#name-then), and [else](https://json-schema.org/draft/2020-12/json-schema-core.html#name-else).
+
+Here's how the validation works in this example:
+
+If the value of `isMember` is true:
+* The then block is applied, which specifies that the `membershipNumber` property should be a string with a minimum length of 10 and a maximum length of 10.
+
+If the value of `isMember` is anything other than true:
+* The else block is applied, which specifies that the `membershipNumber` property can be any string.
+
+```json
+{
+  "$id": "https://example.com/conditional-validation-if-else.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Conditional Validation with If-Else",
+  "type": "object",
+  "properties": {
+    "isMember": {
+      "type": "boolean"
+    },
+    "membershipNumber": {
+      "type": "string"
+    }
+  },
+  "if": {
+    "properties": {
+      "isMember": {
+        "const": true
+      }
+    }
```

Comment:
This needs a `required: [ isMember ]` to work.  Otherwise it'll trigger when `isMember` is absent.